homeassistant/components/slide/__init__.py

```diff
@@ -0,0 +1,159 @@
+                vol.Required(CONF_PASSWORD): cv.string,
+                vol.Optional(
+                    CONF_SCAN_INTERVAL, default=DEFAULT_SCAN_INTERVAL
+                ): cv.time_period,
+                vol.Optional(CONF_TIMEOUT, default=DEFAULT_TIMEOUT): cv.positive_int,
+                vol.Optional(CONF_RETRY, default=DEFAULT_RETRY): cv.positive_int,
+            }
+        )
+    },
+    extra=vol.ALLOW_EXTRA,
+)
+
+
+async def async_setup(hass, config):
+    """Set up the Slide platform."""
+
+    # pylint: disable=unused-argument
+    async def update_slides(now=None):
+        """Update slide information."""
+        result = await hass.data[DOMAIN][API].slidesoverview()
+
+        if result is None:
+            _LOGGER.error("Slide API does not work or returned an error")
+            return
+
+        if result:
+            _LOGGER.debug("Slide API returned %d slide(s)", len(result))
+        else:
+            _LOGGER.warning("Slide API returned 0 slides")
+
+        for key in hass.data[DOMAIN][SLIDES]:
+            hass.data[DOMAIN][SLIDES][key]["pos"] = None
+
+        for slide in result:
+            if "device_id" not in slide:
+                _LOGGER.error(
+                    "Found invalid Slide entry, 'device_id' is " "missing. Entry=%s",
+                    slide,
+                )
+                continue
+
+            uid = slide["device_id"].replace("slide_", "")
+            slidenew = {}
+            slidenew["mac"] = uid
+            slidenew["id"] = slide["id"]
+            slidenew["name"] = slide["device_name"]
+
+            if "device_info" not in slide:
+                _LOGGER.error(
+                    "Slide %s (%s) has no 'device_info' Entry=%s",
+                    slide["id"],
+                    slidenew["mac"],
+                    slide,
+                )
+                continue
+
+            # Check if we have 'pos' (OK) or 'code' (NOK)
+            if "pos" in slide["device_info"]:
+                slidenew["online"] = True
+                slidenew["pos"] = slide["device_info"]["pos"]
+                if slidenew["pos"] < 0:
+                    slidenew["pos"] = 0
+                elif slidenew["pos"] > 1:
+                    slidenew["pos"] = 1
+            elif "code" in slide["device_info"]:
+                slidenew["online"] = False
+                _LOGGER.warning(
+                    "Slide %s (%s) is offline with " "code=%s",
+                    slide["id"],
+                    slidenew["mac"],
+                    slide["device_info"]["code"],
+                )
+            else:
+                slidenew["online"] = False
+                _LOGGER.error(
+                    "Slide %s (%s) has invalid 'device_info'" " %s",
+                    slide["id"],
+                    slidenew["mac"],
+                    slide["device_info"],
+                )
+
+            hass.data[DOMAIN][SLIDES][uid] = slidenew
```

Comment:
If we update the existing dict, and make sure we set a default, we can simplify downstream code.
```suggestion
            old_slide = hass.data[DOMAIN][SLIDES].setdefault(uid, {})
            old_slide.update(slidenew)
```

Reply:
Made the change.